spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -277,31 +287,33 @@ object DistriOptimizer {
 
           parameters.sendWeightPartition()
           Iterator.empty
-        }).count()
+        }.count()
 
-        accumulateCount += recordsNum.value
+        recordsProcessedThisEpoch += recordsNum.value
         val end = System.nanoTime()
         wallClockTime += end - start
         optimMethod.state.update("epoch", driverState[Int]("epoch"))
         optimMethod.state.update("neval", driverState[Int]("neval"))
-        driverState("Loss") = lossSum.value.toFloat / finishedModelNum
+        driverState("Loss") = lossSum.value.toFloat / numFinishedModelUpdates
         optimMethod.state.update("Loss", driverState[Float]("Loss"))
         if (validationMethods.isDefined) {
           optimMethod.state.update("score", driverState[Float]("score"))
         }
         optimMethod.updateHyperParameter()
         driverState("Throughput") = recordsNum.value.toFloat / ((end - start) / 1e9f)
         driverState("LearningRate") = -optimMethod.getLearningRate().toFloat
-        logger.info(s"${_header} Train ${recordsNum.value} in ${(end - start) / 1e9}seconds. " +
-          s"Throughput is ${driverState("Throughput")} records/second. Loss is ${
+        val _header = header(driverState[Int]("epoch"), recordsProcessedThisEpoch, numSamples,
```

Comment:
I moved this down because of what I see as an error in the logging before. When an epoch was finished you might see the logger print out something like:

````
2017-08-01 14:58:03 INFO  DistriOptimizer$:307 - [Epoch 1 8160/8160][Iteration 256][Wall Clock 6.607695635s] Trained 32 records in 0.029281884 seconds. Throughput is 1092.8258 records/second. Loss is 1.6109945E-4.
2017-08-01 14:58:03 INFO  DistriOptimizer$:351 - [Epoch 1 8192/8192][Iteration 256][Wall Clock 6.607695635s] Epoch finished. Wall clock time is 6691.791083 ms
````

The first number was, incorrectly IMO, the number of samples processed _before_ this iteration, not the number of samples that had been currently processed. That also has the consequence of the final message, which says "Epoch finished" indicating that it did not process all of the records e.g. only 8160 out of 8192. After moving this here, it will read correctly.